From e54ea84a2e6ef528e5bc2e8bf905ee6f976f4a6c Mon Sep 17 00:00:00 2001
From: Kody A Kantor <kody.kantor@gmail.com>
Date: Fri, 20 Apr 2018 21:18:08 +0000
Subject: [PATCH] MANTA-3624 create pgstatsmon image Reviewed by: David Pacheco
 <dap@joyent.com> Approved by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>

---
 config/services/pgstatsmon/service.json | 14 ++++++++++++++
 docs/man/man1/manta-adm.md              |  3 +++
 lib/layout.js                           |  7 ++++++-
 lib/services.js                         |  4 +++-
 man/man1/manta-adm.1                    |  3 +++
 test/tst.layout.js.out                  |  8 ++++++++
 test/tst.services.js                    |  7 ++++---
 7 files changed, 41 insertions(+), 5 deletions(-)
 create mode 100644 config/services/pgstatsmon/service.json

diff --git a/config/services/pgstatsmon/service.json b/config/services/pgstatsmon/service.json
new file mode 100644
index 0000000..1e80e68
--- /dev/null
+++ b/config/services/pgstatsmon/service.json
@@ -0,0 +1,14 @@
+{
+	"params": {
+		"networks": ["manta", "admin"],
+		"ram": 2048
+	},
+        "metadata": {
+                "DATABASE_USER": "moray",
+                "VM_TAG_NAME": "manta_role",
+                "VM_TAG_VALUE": "postgres",
+                "NIC_TAG": "manta",
+                "VMAPI_POLL_INTERVAL": "60000",
+                "SCRAPE_INTERVAL": "60000"
+	}
+}
diff --git a/docs/man/man1/manta-adm.md b/docs/man/man1/manta-adm.md
index 709fd56..1900fc3 100644
--- a/docs/man/man1/manta-adm.md
+++ b/docs/man/man1/manta-adm.md
@@ -95,6 +95,9 @@ Services that are part of the Manta application include:
 **postgres**
   PostgreSQL databases used for storing object and job metadata
 
+**pgstatsmon**
+  Monitoring system for Postgres
+
 **reshard**
   Automated resharding system for Manta
 
diff --git a/lib/layout.js b/lib/layout.js
index a90ad70..4c069fe 100644
--- a/lib/layout.js
+++ b/lib/layout.js
@@ -91,7 +91,12 @@ var ML_SERVICES_EXACT = {
 	 * "reshard" is an optional component for systems that are undergoing
 	 * resharding.
 	 */
-	'reshard': 0
+	'reshard': 0,
+
+	/*
+	 * "pgstatsmon" is an optional component for monitoring Postgres.
+	 */
+	'pgstatsmon': 0
 };
 
 /*
diff --git a/lib/services.js b/lib/services.js
index 73eb1c2..91f32a5 100644
--- a/lib/services.js
+++ b/lib/services.js
@@ -51,7 +51,8 @@ var mSvcNames = [
     'madtom',
     'marlin-dashboard',
     'marlin',
-    'reshard'
+    'reshard',
+    'pgstatsmon'
 ];
 
 /*
@@ -74,6 +75,7 @@ var mSvcConfigs = {
     'madtom':		{ 'oneach': true,  'probes': true,  'sharded': false },
     'marlin-dashboard':	{ 'oneach': true,  'probes': true,  'sharded': false },
     'reshard':		{ 'oneach': true,  'probes': true,  'sharded': false },
+    'pgstatsmon':	{ 'oneach': true,  'probes': true,  'sharded': false },
     'marlin':		{ 'oneach': false, 'probes': false, 'sharded': false },
     'propeller':	{ 'oneach': true,  'probes': false, 'sharded': false }
 };
diff --git a/man/man1/manta-adm.1 b/man/man1/manta-adm.1
index 7377c21..61f41f8 100644
--- a/man/man1/manta-adm.1
+++ b/man/man1/manta-adm.1
@@ -91,6 +91,9 @@ Manages asynchronous operations like garbage collection, metering, and auditing
 \fBpostgres\fP
 PostgreSQL databases used for storing object and job metadata
 .TP
+\fBpgstatsmon\fP
+Monitoring system for Postgres
+.TP
 \fBreshard\fP
 Automated resharding system for Manta
 .TP
diff --git a/test/tst.layout.js.out b/test/tst.layout.js.out
index 34d0a07..8e59874 100644
--- a/test/tst.layout.js.out
+++ b/test/tst.layout.js.out
@@ -229,6 +229,7 @@ summary:
      marlin-dashboard     -                1
      marlin               -               32
      reshard              -                 
+     pgstatsmon           -                 
      postgres             1                3
      postgres             2                3
      postgres             3                3
@@ -610,6 +611,7 @@ summary:
      marlin-dashboard     -                1
      marlin               -               16
      reshard              -                 
+     pgstatsmon           -                 
      postgres             1                3
      moray                1                3
 --------------------------------------------------
@@ -720,6 +722,7 @@ summary:
      marlin-dashboard     -                1
      marlin               -               16
      reshard              -                 
+     pgstatsmon           -                 
      postgres             1                3
      postgres             2                3
      postgres             3                3
@@ -1035,6 +1038,7 @@ summary:
      marlin-dashboard     -                1
      marlin               -               48
      reshard              -                 
+     pgstatsmon           -                 
      postgres             1                3
      postgres             2                3
      moray                1                3
@@ -1518,6 +1522,7 @@ summary:
      marlin-dashboard     -                1
      marlin               -               96
      reshard              -                 
+     pgstatsmon           -                 
      postgres             1                3
      postgres             2                3
      postgres             3                3
@@ -1730,6 +1735,7 @@ summary:
      marlin-dashboard     -                                                  1
      marlin               -               16               16               16
      reshard              -                                                   
+     pgstatsmon           -                                                   
      postgres             1                1                1                1
      moray                1                1                1                1
 --------------------------------------------------
@@ -1955,6 +1961,7 @@ summary:
      marlin-dashboard     -                                                  1
      marlin               -               16               16               16
      reshard              -                                                   
+     pgstatsmon           -                                                   
      postgres             1                1                1                1
      postgres             2                1                1                1
      moray                1                1                1                1
@@ -2421,6 +2428,7 @@ summary:
      marlin-dashboard     -                                                  1
      marlin               -               48               48               48
      reshard              -                                                   
+     pgstatsmon           -                                                   
      postgres             1                1                1                1
      postgres             2                1                1                1
      postgres             3                1                1                1
diff --git a/test/tst.services.js b/test/tst.services.js
index f86771f..c927c79 100644
--- a/test/tst.services.js
+++ b/test/tst.services.js
@@ -39,7 +39,8 @@ var knownServices = [
     'marlin-dashboard',
     'marlin',
     'propeller',
-    'reshard'
+    'reshard',
+    'pgstatsmon'
 ];
 
 function main()
@@ -91,8 +92,8 @@ function main()
 	/*
 	 * Test serviceSupportsProbes().
 	 */
-	assertplus.deepEqual([ 'marlin', 'propeller' ], knownServices.filter(
-	    function (svcname) {
+	assertplus.deepEqual([ 'marlin', 'propeller' ],
+	    knownServices.filter(function (svcname) {
 		return (!services.serviceSupportsProbes(svcname));
 	    }));
 	assertplus.deepEqual(services.mSvcNamesProbes,
-- 
2.21.0

